File: SaviourBackend/src/main/java/com/org/SaviourBackend/service/ServicesService.java
package com.org.SaviourBackend.service;

import java.util.List;

import com.org.SaviourBackend.dto.ServicesDto;

public interface ServicesService {
	
//	Create service
	ServicesDto createService(ServicesDto servicesDto);
	
// Retrieve all services
	List<ServicesDto> getAllServices();
	
//	Retrieve Service by ID
	ServicesDto getServiceById(Long serviceId);
	
//	Update Services
	ServicesDto updateService(Long serviceId, ServicesDto updateService);
	
//	Delete Services
	void deleteService(Long serviceId);
}

File: frontend/saviour-billing-portal/src/services/ServicesService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/services";

export const createService = (services) => axios.post(REST_API_BASE_URL, services);

export const listServices = () => axios.get(REST_API_BASE_URL);

export const deleteService = (serviceId) => axios.delete(REST_API_BASE_URL + '/' +serviceId);

export const getService = (serviceId) => axios.get(REST_API_BASE_URL+"/"+serviceId);

export const updateService = (serviceId, service) => axios.put(REST_API_BASE_URL+"/"+serviceId, service);